src/xGadgetron/cGadgetron/tests/CMakeLists.txt

```diff
@@ -17,13 +17,77 @@
 #
 #=========================================================================
 
+
+function(get_zenodo_file path url md5)
+        message(STATUS "Downloading (if not already done)...
+                Path:     ${path}
+                URL:      ${url}
+                MD5:      ${md5}")
+        file(DOWNLOAD "${url}" "${path}" SHOW_PROGRESS EXPECTED_MD5 "${md5}")
+endfunction()
+
+OPTION(DOWNLOAD_ZENODO_TEST_DATA "download zenodo data for tests" ON)
+if (DOWNLOAD_ZENODO_TEST_DATA)
+        set(MR_RPE_DATA "${CMAKE_SOURCE_DIR}/data/examples/MR/zenodo/3D_RPE_Lowres.h5")
+        get_zenodo_file(
+                "${MR_RPE_DATA}"
+                "https://zenodo.org/record/4600937/files/3DRPE_Lowres.h5?download=1"
+                "a692ba3a6c456d1ad49750717a9d5dda")
+endif()
+
+OPTION(RERUN_ISMRMRD_SHEPP_LOGAN "Generate testdata compatible with ISMRMRD Version" ON)
+find_program(EXE_ISMRMRD_SL NAMES ismrmrd_generate_cartesian_shepp_logan
+                NAMES_PER_DIR
+                PATHS ${CMAKE_INSTALL_PREFIX})
+
+
+set(TESTDATA_OUTPUT_PATH "${SIRF_SOURCE_DIR}/data/examples/MR/")
+set(TESTDATA_OUTPUT_NAME_FULL "simulated_MR_2D_cartesian.h5")
+set(TESTDATA_OUTPUT_NAME_GRAPPA "simulated_MR_2D_cartesian_Grappa2.h5")
+
+set(TESTDATA_TEMP_PREFIX "testdata_tmp")
+
+if(EXE_ISMRMRD_SL AND RERUN_ISMRMRD_SHEPP_LOGAN)
+    add_custom_target(GENERATE_SIMULATED_TESTDATA ALL
+        COMMAND rm -f ${TESTDATA_OUTPUT_NAME_FULL}
+        COMMAND rm -f ${TESTDATA_OUTPUT_NAME_GRAPPA}
+        COMMAND env LD_LIBRARY_PATH=${CMAKE_INSTALL_PREFIX}/lib:$LD_LIBRARY_PATH ${EXE_ISMRMRD_SL} -r 2 -o "${TESTDATA_TEMP_PREFIX}_1.h5"
+        COMMAND env LD_LIBRARY_PATH=${CMAKE_INSTALL_PREFIX}/lib:$LD_LIBRARY_PATH ${EXE_ISMRMRD_SL} -a 2 -o "${TESTDATA_TEMP_PREFIX}_2.h5"
+        WORKING_DIRECTORY ${TESTDATA_OUTPUT_PATH})
+elseif(RERUN_ISMRMRD_SHEPP_LOGAN)
+    message(ERROR "We could not find the ISMRMRD executable for the Shepp Logan Generation." )
+else()
+    add_custom_target(GENERATE_SIMULATED_TESTDATA ALL
+        COMMAND "")
+endif()
+
 add_library(MR_TESTS_CPP_AUXILIARY ${CMAKE_CURRENT_SOURCE_DIR}/mrtest_auxiliary_funs.cpp)
 target_link_libraries(MR_TESTS_CPP_AUXILIARY PUBLIC csirf cgadgetron)
 
+add_executable(MR_PROCESS_TESTDATA ${CMAKE_CURRENT_SOURCE_DIR}/mrtests_prep_testdata.cpp)
+add_dependencies(MR_PROCESS_TESTDATA GENERATE_SIMULATED_TESTDATA)
+target_link_libraries(MR_PROCESS_TESTDATA PUBLIC MR_TESTS_CPP_AUXILIARY csirf cgadgetron)
+
+if(RERUN_ISMRMRD_SHEPP_LOGAN)
+        add_custom_target(PROCESS_SIMULATED_TESTDATA ALL
```

Comment:
@KrisThielemans, as you mentioned [here](https://github.com/SyneRBI/SIRF/pull/881/files/1257f2fb1e01f730fb3605365d4b6a5ad0f00c67#r642557433) it is indeed a runtime error, since I run the executable during the build.

The logic was initially:
* problem: the ISMRMRD-simulated Shepp-Logan in the data repository is outdated, and the XML version in the header can lead to the program rejecting the test data as input
* solution: I run the Shepp-Logan simulation from `CMake` during the build to generate the test data on the fly
* next problem: then the geometry fields in the acquisitions were not populated, so we got all these warnings that the orientation vectors are zero.
* next solution: @evgueni-ovtchinnikov and I had written code that populates these fields after the simulation had finished, and I compiled this into an executable and run it after I generated the test data on the fly
* next problem: during this run it does not find this gadgetron library that I never assumed it would need.

Since the above first problem was now apparently solved by updating the data repository, I could just delete the part that re-runs the SheppLogan simulation and everything that follows, since it looks like we don't need it after all.

Reply:
ok. thanks for that. This explains it all.

You run a SIRF executable during the build, but that depends on `LD_LIBRARY_PATH` etc being set correctly. But this is set via env_sirf.sh, which on GHA is only done after the build. On your local machine, you more than likely have it set before you build, so you don't see this problem.

creating an issue...

Reply:
https://github.com/SyneRBI/SIRF-SuperBuild/issues/546

Reply:
@johannesmayer what about editing
https://github.com/SyneRBI/SIRF/blob/0349fe6d16b8bb268378c5c37b2a4dee9cb55f0a/.github/workflows/build-test.yml#L67-L68
to insert `source ~/install/env_sirf.sh` before the build?

Reply:
I tried but it would not accept my access token when I tried to push to my PR branch after I modified the workflow....

Reply:
?? There is nothing magical about that file, so maybe you had a temporary problem.

I'm assuming this problem is unresolved. If so, can you create a new issue or PR addressing only this problem? Otherwise we will just forget until it hits us again.